docs/Stack/Balanced-parentheses-checker.md

```diff
@@ -0,0 +1,103 @@
+---
+id: balanced-parentheses-checker
+title: Balanced-parentheses-checker 
+sidebar_label: Balanced-parentheses-checker 
+description: "This algorithm checks if the parentheses in a given expression are balanced using a stack to ensure each opening parentheses has a corresponding closing one."  
+tags: [dsa, algorithms, stack]
+---
+
+### Problem Definition
+The goal of this program is to check if an expression containing parentheses is balanced. The expression may contain three types of parentheses:
+- Round brackets ()
+- Curly brackets {}
+- Square brackets []
+                                        
+An expression is said to be balanced if every opening parenthesis has a corresponding closing parenthesis in the correct order.
+
+### Approach
+This program uses a stack data structure to solve the problem of checking whether the parentheses in an expression are balanced. The stack is ideal for this task as it allows tracking unmatched opening parentheses, which are then compared with the closing parentheses.
+
+### Algorithm Steps
+Scan the expression:
+1. If an opening parenthesis ((, {, or [) is encountered, push it onto the stack.
```

Comment:
Our documentation formatting is based on Markdown (`.md` / `.mdx`) plus React (`.js` / `.jsx`). As a result, we can't use `{ }` directly because the curly braces are interpreted as code, not as text.

Reply:
done